src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -168,84 +222,136 @@ public void setNumberOfInstances(int numberOfInstances) {
 			v.setSpaceId(tuple.getT2());
 			return Mono.just(v);
 		});
-		
-		Flux<Map<String, SpaceApplicationSummary>> spaceSummaryFlux = osaVectorSpaceFlux.flatMapSequential(v -> this.getSpaceSummary(v.getSpaceId()));
+
+		Flux<Map<String, SpaceApplicationSummary>> spaceSummaryFlux = osaVectorSpaceFlux
+				.flatMapSequential(v -> this.getSpaceSummary(v.getSpaceId()));
 		Flux<OSAVector> osaVectorApplicationFlux = Flux.zip(osaVectorSpaceFlux, spaceSummaryFlux).flatMap(tuple -> {
 			OSAVector v = tuple.getT1();
-			
+
 			if (INVALID_SUMMARY == tuple.getT2()) {
 				// NB: This drops the current target!
 				return Mono.empty();
 			}
-			
+
 			Map<String, SpaceApplicationSummary> spaceSummaryMap = tuple.getT2();
-			SpaceApplicationSummary sas = spaceSummaryMap.get(v.getTarget().getApplicationName().toLowerCase(LOCALE_OF_LOWER_CASE_CONVERSION_FOR_IDENTIFIER_COMPARISON));
+			SpaceApplicationSummary sas = spaceSummaryMap.get(v.getTarget().getApplicationName()
+					.toLowerCase(LOCALE_OF_LOWER_CASE_CONVERSION_FOR_IDENTIFIER_COMPARISON));
 			/*
-			 * Due to https://github.com/cloudfoundry/cloud_controller_ng/issues/1523,
-			 * we cannot rely on sas.getId() (i.e. it may contain wrong information)
+			 * Due to https://github.com/cloudfoundry/cloud_controller_ng/issues/1523, we
+			 * cannot rely on sas.getId() (i.e. it may contain wrong information)
 			 */
-			
+
 			if (sas == null) {
 				// NB: This drops the current target!
 				return Mono.empty();
 			}
 
 			List<String> urls = sas.getUrls();
 			if (urls != null && !urls.isEmpty()) {
-				v.setAccessURL(this.determineAccessURL(v.getTarget().getProtocol(), urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns(), v.getTarget().getPath()));
+				// Set the access url to the selected route (without any protocol or path yet)
+				v.setAccessURL(this.determineApplicationRoute(urls,
+						v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));
+			}
+
+			// In the interest of backwards compatibility lest do this inside a try.
+			// There is little reason why this should not find the correct domain
+			try {
+				Route route = sas.getRoutes().stream().filter(rt -> v.getAccessURL().startsWith(rt.getHost()+"."+rt.getDomain().getName())).findFirst().get();
+				v.setDomainId(route.getDomain().getId());	
+			} catch (Exception e) {
+				log.warn(String.format("unable to find matching domain for the url %s", v.getAccessURL()));
 			}
 			
 			v.setNumberOfInstances(sas.getInstances());
-			
+
 			return Mono.just(v);
 		});
-		
+
+		Flux<List<DomainResource>> domainFlux = osaVectorApplicationFlux.flatMapSequential(v -> {
+			return this.cfAccessor.retrieveAllDomains(v.getOrgId()).flatMap(mapper -> Mono.just(mapper.getResources()));
```

Comment:
might want to do 
```suggestion
			return this.cfAccessor.retrieveAllDomains(v.getOrgId()).map(mapper -> mapper.getResources());
```
instead (mainly just easier to read).